pyccl/halos/massdef.py

```diff
@@ -246,69 +246,43 @@ def from_name(cls, name):
         Returns:
             MassDef subclass corresponding to the input name.
         """
-        mass_defs = {m.name: m for m in cls.__subclasses__()}
-
-        if name in mass_defs:
-            return mass_defs[name]
-        else:
+        try:
+            return eval(f"MassDef{name.capitalize()}")
```

Comment:
why capitalize?

Reply:
`capitalize()` capitalizes only the first letter of every word, if it starts with a letter. Basically there for `MassDefVir`.